src/client/ua_client.c

```diff
@@ -633,11 +646,11 @@ UA_StatusCode UA_Client_disconnect(UA_Client *client) {
         return UA_STATUSCODE_BADNOTCONNECTED;
     UA_StatusCode retval = UA_STATUSCODE_GOOD;
     /* Is a session established? */
-    if(client->channel.connection->state == UA_CONNECTION_ESTABLISHED &&
+    if(client->channel->connection->state == UA_CONNECTION_ESTABLISHED &&
```

Comment:
Why not client->connection->state?
Multiple levels of indirection are baaaad.
If we always go via the channel, we wouldn't need the extra pointer for the connection.